### DISCUSSION TITLE
How to override Stripe Class

### DISCUSSION THREAD

**Discussion by zimbar** [OFFLINE-GmbH/oc-mall-plugin#1164] [Q&A]
Hy

I use stripe for payment, but i have to change some specific logic in  class OFFLINE\Mall\Classes\Payments\Stripe. Can you tel me how can i override this class in a custom plugin.

Thanks

**Comment by tobias-kuendig** [comment]
Copy the original Stripe class to your plugin and register it as a new payment provider. The docs tell you how to do that:
https://offline-gmbh.github.io/oc-mall-plugin/development/core/payment-providers.html#register-a-paymentprovider

This is easier than trying to change the built-in class. This way, you take full control over all logic.

**Comment by zimbar** [comment]
Thanks, i will do that
